src/v/raft/heartbeat_manager.cc

```diff
@@ -307,14 +307,23 @@ void heartbeat_manager::process_reply(
             continue;
         }
         vlog(hbeatlog.trace, "Heartbeat reply from node: {} - {}", n, m);
```

Comment:
The checking being added looks good to protect against crashing, but should we also validate the message's node ID against the value of `n` here?  It sounds like in #4623 the node ID was clearly invalid before we even got to the group ID, so that's an even clearer signal for "this message is garbage" if we see it in the field.